DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -80,18 +82,67 @@ GROUP BY
             ).FirstOrDefault();
         }
 
+        public bool DoesProgressExist(int candidateId, int customisationId)
+        {
+            return connection.Query<int>(
+                @"SELECT ProgressId
+                        FROM Progress
+                        WHERE CandidateID = @candidateId
+                          AND CustomisationID = @customisationId
+                          AND SystemRefreshed = 0
+                          AND RemovedDate IS NULL",
+                new { candidateId, customisationId }
+            ).Any();
+        }
+
         public int GetProgressId(int candidateId, int customisationId)
         {
-            // TODO HEEDLS-202: change QueryFirstOrDefault to creating progress record if not found
             return connection.QueryFirstOrDefault<int>(

```

Comment:
As commented before we should not use QueryFirstOrDefault. This will return progress id 0 if a progress id can't be found which is not what we want.